R/utils.R

```diff
@@ -259,82 +259,500 @@ rowcol_to_dim <- function(row, col) {
+#'    select the data columns 4, 5, 6, 7 of `mtcars` + column names
+#' 8. provide the position of a single column by name `wb_dims(x = mtcars, cols = "mpg")`.
+#' 9. provide a row span with a column. `wb_dims(x = mtcars, cols = "mpg", rows = 5:22)`
+#'
+#' To reuse, a good trick is to create a wrapper function, so that styling can be
+#' performed seamlessly.
+#'
+#' ```R
+#' wb_dims_cars <- function(...) {
+#'   wb_dims(x = mtcars, from_row = 2, from_col = "B", ...)
+#' }
+#' # using this function
+#' wb_dims_cars() # full grid (data + column names)
+#' wb_dims_cars(select = "data") # data only
+#' wb_dims_cars(select = "col_names") # select column names
+#' wb_dims_cars(cols = "vs") # select the `vs` column
+#' ```
+#'
+#' It can be very useful to apply many rounds of styling sequentially.
+#'
+#'
+#' @details
+#'
+#' #' When using `wb_dims()` with an object, the default behavior is
+#' to select only the data / row or columns in `x`
+#' If you need another behavior, use `wb_dims()` without supplying `x`.
+#'
+#' * `x` An object (typically a `matrix` or a `data.frame`, but a vector is also accepted.)
+#' * `from_row` / `from_col` the starting position of `x`
+#'   (The `dims` returned will assume that the top left corner of `x` is at `from_row / from_col`
+#' * `rows` Optional Which row span in `x` should this apply to.
+#'   If `rows` = 0, only column names will be affected.
+#' * `cols` a range of columns id in `x`, or one of the column names of `x`
+#'   (length 1 only accepted for column names of `x`.)
+#' * `row_names` A logical, this is to let `wb_dims()` know that `x` has row names or not.
+#'   If `row_names = TRUE`, `wb_dims()` will increment `from_col` by 1.
+#' * `col_names` `wb_dims()` assumes that if `x` has column names, then trying to find the `dims`.
+#'
+#'
+#' `wb_dims()` tries to support most possible cases with `row_names = TRUE` and `col_names = FALSE`,
+#' but it works best if `x` has named dimensions (`data.frame`, `matrix`), and those parameters are not specified.
+#'  data with column names, and without row names. as the code is more clean.
+#'
+#' In the `add_data()` / `add_font()` example, if writing the data with row names
+#'
+#' @param ... construct `dims` arguments, from rows/cols vectors or objects that
+#'   can be coerced to data frame. `x`, `rows`, `cols`, `from_row`, `from_col`,
+#'   `row_names`, and `col_names` are accepted.
+#' @param select A string, one of the followings.
+#'    it improves the selection of various parts of `x`
+#'    One of "x", "data", "col_names", or "row_names".
+#'    `"data"` will only select the data part, excluding row names and column names  (default if `cols` or `rows` are specified)
+#'    `"x"` Includes the data, column and row names if they are present. (default if none of `rows` and `cols` are provided)
+#'    `"col_names"` will only return column names
+#'    `"row_names"` Will only return row names.
+#'
+#' @return A `dims` string
+#' @export
```

Comment:
The examples are a bit out of hand. In a follow up we can move them to a vignette (openxlsx2.Rmd) or something else.

Reply:
I tried some reorg, but could always be done in a follow-up.